packages/updating-element/src/updating-element.ts

```diff
@@ -534,6 +537,26 @@ export abstract class UpdatingElement extends HTMLElement {
    */
   private _reflectingProperty: PropertyKey | null = null;
 
+  /**
+   * Set of callbacks called in connectedCallback.
+   */
+  connectedCallbacks: Set<connectCallback> = new Set();
+
+  /**
+   * Set of callbacks called in disconnectedCallback.
+   */
+  disconnectedCallbacks: Set<connectCallback> = new Set();
+
+  /**
+   * Set of callbacks called before update.
+   */
+  updateCallbacks: Set<updateCallback> = new Set();
+
+  /**
+   * Set of callbacks called after updated.
+   */
+  updatedCallbacks: Set<updateCallback> = new Set();
```

Comment:
Rather than four sets of callbacks I'd be interested in seeing a single `addController()` API that tears off the callbacks at that time, similar to `customElements.define()`

Reply:
Using a single set now.